src/Containers/Microsoft.NET.Build.Containers/Layer.cs

```diff
@@ -44,111 +45,78 @@ public static Layer FromDescriptor(Descriptor descriptor)
     }
 
     public static Layer FromDirectory(string directory, string containerPath, bool isWindowsLayer)
```

Comment:
I've removed the `FromFiles` method. The implementation enumerates recursively over both directories and files using `FileSystemEnumerable` which doesn't work for `FromFiles`.

Using `FileSystemEnumerable` we will encounter parent directories before their children which gives a straight-forward way to add the parent directory entries to the archive.